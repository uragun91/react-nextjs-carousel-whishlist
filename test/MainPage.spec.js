import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import App from "../pages/mainpage";

describe("With Enzyme", () => {
  it('Total 5 div are using for Main Layout component"', () => {
    const app = shallow(<App />);

    expect(app.find('div')).toHaveLength(5)
  });
});

describe("With Enzyme", () => {
  it('3 h4 is using for text', () => {
    const app = shallow(<App />);

    expect(app.find('h4')).toHaveLength(3)
  });
});

describe("With Enzyme", () => {
  it('Heading displaying with specific text', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.text().includes('Popular')).toBe(true);
    expect(wrapper.text().includes('Top Rated')).toBe(true);
    expect(wrapper.text().includes('Upcoming')).toBe(true);
  });
});


describe("With Snapshot Testing", () => {
  it('Application manageing movies and detail of specific movie.!"', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});