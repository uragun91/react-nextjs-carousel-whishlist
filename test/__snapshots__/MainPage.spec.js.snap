// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows Future-proof your business. WiFi 6E is here.!" 1`] = `
<div
  className="container"
>
  <div
    className="vertical-center"
  >
    <h4>
      Popular
    </h4>
    <div
      className="slider"
    >
      <div
        className="slides"
      />
    </div>
    <h4>
      Top Rated
    </h4>
    <div
      className="slider"
    >
      <div
        className="slides"
      />
    </div>
    <h4>
      Upcoming
    </h4>
    <div
      className="slider"
    >
      <div
        className="slides"
      />
    </div>
  </div>
</div>
`;

exports[`With Snapshot Testing Application manageing movies and detail of specific movie.!" 1`] = `
<div
  className="container"
>
  <div
    className="vertical-center"
  >
    <h4>
      Popular
    </h4>
    <div
      className="slider"
    >
      <div
        className="slides"
      />
    </div>
    <h4>
      Top Rated
    </h4>
    <div
      className="slider"
    >
      <div
        className="slides"
      />
    </div>
    <h4>
      Upcoming
    </h4>
    <div
      className="slider"
    >
      <div
        className="slides"
      />
    </div>
  </div>
</div>
`;
